### PR TITLE
issue-1681/multi-day-events-consistency (month view)

### PR DESCRIPTION
**Pull request by river-bbc**
## Description
This PR improves displaying of multi day events in the calendar month view

## Screenshots
**Month view**
Screenshot 2023-12-30 at 14 29 08

## Changes

* Month Calendar View
  * Displays multi day events on each day they occur
    * Shows the date and time as  "All day today" when a multi day event spans an entire day
    * This also works with the checkbox: checking a multi event selects instances of that event! It's cool. 

## Notes to reviewer
Add some multi day events at different times, also across weeks and months, and check they display as expected

## Related issues
Part of #1681

### CODE REVIEW

**Comment by joe-irving** [comment]
Just on this, appreciate how quick this was picked up. From my perspective getting this as it is merged would be great. 

I am bring a group on board and in the process of getting them to use events, most of which are multi day.

Not sure how long this would normally take but I figured a comment to bump it wouldn't be a bad thing

**Comment by ziggabyte** [comment]
@river-bbc hey do you feel like continuing working on this, or should we have someone else pick up where you ended things? :D

File: src/features/calendar/components/utils.ts [hunk 1]
```diff
@@ -3,12 +3,30 @@ import dayjs from 'dayjs';
...
 
+export function isAllDay(start: string, end: string): boolean {
+  const startDate = new Date(removeOffset(start));
+  const endDate = new Date(removeOffset(end));
+
+  // Check if the start and end dates are not on the same day
+  if (startDate.toDateString() !== endDate.toDateString()) {
+    // If start time and end time are 00:00:00 return true
+    if (
+      startDate.toString().split(' ')[4] == '00:00:00' &&
```
Comment (henrycatalinismith) [inline src/features/calendar/components/utils.ts:21]: This is a little unrelated as your PR only moves this function from one file to another, but I'm curious why this function works and maybe it's an opportunity for me to learn about some of the horrible grubby little details of how Zetkin handles dates.

An all-day event in Sweden for e.g. today would have a UTC start date of  2024-03-15 23:00:00 and an end date of 2024-03-16 23:00:00. So the `startDate.toString().split(' ')[4] == '00:00:00'` check would return `false `and as a result the event would be mis-classified as not an all-day event.

Struggling to figure out independently why this isn't an issue with Zetkin.

**Comment by henrycatalinismith** [comment]
We've spent an hour or two testing and debugging and tweaking this together. It's a net improvement on the status quo. Seems like there's definitely going to be more iterations on this and to add functionality and fix bugs but it's so valuable to be able to see the multiday events that it's definitely better to have this merged already.

**Review by henrycatalinismith — APPROVED** [review]

**Pull request merged**